props/saber_BC_buttons.h

```diff
@@ -1319,29 +1319,33 @@ struct BCSelectBladeMode : public SPEC::MenuBase {
   uint16_t size() override { return NUM_BLADES; }
 
   void mode_activate(bool onreturn) override {
-    mode::getSL<SPEC>()->SaySelectBlade();
-    SPEC::SteppedMode::mode_activate(onreturn);
-    highlighted_blade_.Start(current_blade());
-    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+    // Just exit after setting length
+    if (onreturn) {
+      popMode(); 
```

Comment:
I think this is fine, but there is an alternative that may potentially be simpler. Basically, you could do popMode() right before pushing BladeLengthMode on the mode stack. That way this mode won't be in the mode stack anymore, and we won't need to do anything special when we return from BladeLengthMode.

Reply:
ah. Works great, thanks.